typescriptFixme.ts

```diff
@@ -57,59 +66,81 @@ class Point {
 }
 
 // annotate `PointPerson` to use all properties of `Person` and then add the missing required property
-class PointPerson {
-  constructor(name) {
+class PointPerson implements Person {
+  name: string;
+  move(): void {}
+
+
+  constructor(name: string) {
     this.name = name;
   }
 }
 
 // Inheritance
 // Make `Point3D` inherit from `Point`
 // Don't forget to call the parent constructor
-class Point3D {
-  constructor(x, y, z = 0) {
+class Point3D extends Point{
+  z: number;
+
+
+  constructor(x: number, y: number, z: number = 0) {
+    super(x,y);
     this.z = z;
   }
-  dist() {
-    var d = super.dist();
+
+  dist(): number {
+    const d = super.dist();
     return Math.sqrt(d * d + this.z * this.z);
   }
 }
 
 // Make `Tuple` Generic that it accepts to generic types as arguments
 
-class Tuple {
-  constructor(item1, item2) {
+class Tuple <T1, T2> {
+  item1: T1;
+  item2: T2;
+
+  constructor(item1: T1, item2: T2) {
     this.item1 = item1;
     this.item2 = item2;
   }
 }
 
 // create a generic interface called Pair that accepts one generic type for two properties named item1 and item2
 
+interface Pair<T> {
+  item1: T;
+  item2: T;
+}
+
 // create a function that will take a pair with a generic argument and return a `Tuple`
-function pairToTuple(pair) {
+function pairToTuple<T>(pair: Pair<T>): Tuple<T,T> {
   return new Tuple(pair.item1, pair.item2);
 }
 
 var tuple = pairToTuple({ item1: "hello", item2: "world" });
 
 // annotate `make`, `model`, and `year` to be not mutable (read only) outside the constructor
 class Car {
-  make;
-  model;
-  year = 2018;
+  readonly make: string;
+  readonly model: string;
+  readonly year: number = 2018;
 
-  constructor() {
+  constructor(make: string, model: string) {
     this.make = "Unknown Make";
     this.model = "Unknown Model";
   }
 }
 
 // annotate this array
-var arrayOfAnyType = [1, "string", false];
+var arrayOfAnyType: any[] = [1, "string", false];
```

Comment:
don't use type any. declare this type to an array that accepts the specific primitive types using union